detections/endpoint/ssa___rare_parent_process_relationship_lolbas.yml

```diff
@@ -0,0 +1,86 @@
+name: Rare Parent-Child Process Relationship
+id: cf090c78-bcc6-11eb-8529-0242ac130003
+version: 1
+date: '2021-05-20'
+author: Peter Gael, Splunk; Ignacio Bermudez Corrales, Splunk
+type: streaming
+datamodel: []
+description: An attacker may use LOLBAS tools spawned from vulnerable applications
+  not typically used by system administrators. This search leverages the Splunk Streaming
+  ML DSP plugin to find rare parent/child relationships. The list of application has
+  been extracted from https://github.com/LOLBAS-Project/LOLBAS/tree/master/yml/OSBinaries
+search: '| from read_ssa_enriched_events() | eval timestamp=parse_long(ucast(map_get(input_event,
+  "_time"), "string", null)) | eval parent_process=lower(ucast(map_get(input_event,
+  "parent_process_name"), "string", null)), parent_process_name=mvindex(split(parent_process,
+  "\\"), -1), process_name=lower(ucast(map_get(input_event, "process_name"), "string",
+  null)), cmd_line=ucast(map_get(input_event, "process"), "string",
+  null), dest_user_id=ucast(map_get(input_event, "dest_user_id"), "string", null),
+  dest_device_id=ucast(map_get(input_event, "dest_device_id"), "string", null)
+  | where parent_process_name!=null
+  | select parent_process_name, process_name, cmd_line, timestamp, dest_device_id, dest_user_id
+  | conditional_anomaly conditional="parent_process_name" target="process_name"
+  | where (process_name="powershell.exe" OR process_name="regsvcs.exe"
```

Comment:
@peter-cg do we end up reusing this same list of process names in many detections would it make sense to create a macro for this whitelist at some point?

Reply:
It should be in a lookup when that's possible, yeah.